### ISSUE TITLE
HeatExchanger:FluidToFluid problem with dual deadband control

### ISSUE THREAD

**Issue by EnergyArchmage** [NREL/EnergyPlus#5258]
HeatExchanger:FluidToFluid with Control Type of DualDeadbandSetpointModulated does not work very well.  The logic is such that it will only transfer heat when conditions are such that it can succeed at meeting the hi/lo setpoints.  In cases where the demand side inlet temperature is such that, even though the setpoint cannot be completely reached, there could be some beneficial heating or cooling if only it would operate, but it does not.  

The logic is overly restrictive and doesn’t allow the HX to run often enough.  The following lines need to be changed, in PlantHeatExchangerFluidToFluid.cc line 1236:

```
    if ( ( DeltaTCooling > FluidHX( CompNum ).TempControlTol ) && ( DeltaTCoolSetPointSupply < ( -1.0 * FluidHX( CompNum ).TempControlTol ) ) && ( DeltaTCoolSetPointDemand > FluidHX( CompNum ).TempControlTol ) ) {
```

And about line 1248:

```
   } else if ( ( DeltaTHeating > FluidHX( CompNum ).TempControlTol ) && ( DeltaTHeatSetPointDemand < ( -1.0 * FluidHX( CompNum ).TempControlTol ) ) && ( DeltaTHeatSetPointSupply > FluidHX( CompNum ).TempControlTol ) ) {
```

It appears that the analogous logic currently being used for DualDeadbandSetpointOnOff would work fine and modulated should be changed to match.

```
 if ( ( DeltaTCooling > FluidHX( CompNum ).TempControlTol ) && ( SetPointTempHi < FluidHX( CompNum ).SupplySideLoop.InletTemp ) ) {

 } else if ( ( DeltaTHeating > FluidHX( CompNum ).TempControlTol ) && ( SetPointTempLo > FluidHX( CompNum ).SupplySideLoop.InletTemp ) ) {
```

**Comment by Myoldmopar** [comment]
Any relation at all to #4959?

**Comment by EnergyArchmage** [comment]
Added a defect file to dev support repository.  EnergyPlusDevSupport\DefectFiles\5000s\5258. 
Took e+ example file and changed hi setpoint to be 1.0 C lower that demand side inlet temperature so that hi temp setpoint cannot quite be met. 

before plot

![defectfilebeforefix](https://cloud.githubusercontent.com/assets/8754769/11305412/f5a27292-8f82-11e5-80da-16c57ecdba0b.png)

after fix plot

![defectfileafterfix](https://cloud.githubusercontent.com/assets/8754769/11305421/fcb8856c-8f82-11e5-8eeb-950cee7cf9f8.png)

**Comment by mjwitte** [comment]
Closed via #5333
